dts/arm/nordic/nrf52840.dtsi

```diff
@@ -54,6 +54,25 @@
 			label = "UART_1";
 		};
 
+		gpio0: gpio@50000000 {
+			compatible = "nordic,nrf5-gpio";
+			gpio-controller;
+			interrupts = <6 5>;
```

Comment:
Is there a reason to put GPIO interrupts on other priority than everything else?

Also, according to spec the GPIO block cannot issue an interrupt: This is done by GPIOTE block, which is not present in the dtsi file. IMHO we should:
- Add gpiote instance to the dts and place interrupt there.
- Add reference from gpio instance to gpiote instance.

Reply:
Hi, Thanks for reviewing.

The interrupt priority is exactly what the GPIO driver currently uses as default; this was deliberate to reduce friction in introducing this to dts.

The driver code for the GPIO and GPIOTE are in the same module. Are you asking that just an extra gpiote node should be added to the dts (to be used in existing driver), or that the driver also needs to be split up?

I'm new to zephyr, and the way dts is used is still unfamiliar, but I think that using dts rather than kconfig options is good, and adding dts gpio support is a necessary start for many other drivers that used gpios/pins (pwm,i2c,spi,uart etc) - however as I'm not very knowledgeable yet I'm very much just keeping status quo. Any extra advice is appreciated.